src/chapters/unit-testing/unit-testing-in-action.rst

```diff
@@ -0,0 +1,218 @@
+
+- Case should be considered, and
+- whitespace should be considered.
+
+The definition of a palindrome differs sometimes on these two matters, so it's important to test them.
+
+Testing with input ``"abA"``` ensures that case is considered, since the lowercase version of this string, ``"aba"``, *is* a palindrome. Testing with ``"so many dynamos"`` ensures that whitespace is considered, since the version of this string with whitespace removed, ``"somanydynamos"``, *is* a palindrome.
+
+.. note:: It's important to isolate your test cases. For example, ``"So Many Dynamos"`` is a poor choice of input for a negative test, since it contains *two* characteristics that are being tested for, case and the presence of whitespace. If a test with this input failed, it would NOT be clear why it failed.
+
+Including specific tests that demonstrate how *our* ``isPalindrome`` function behaves in these situations helps make our code *self-documenting*. Someone can read our tests and easily see that we *do* consider case and whitespace.
+
+Let's add some test for these negative cases. Add these within the ``describe`` call.
+
+.. sourcecode:: js
+
+   it("should return false for a longer non-palindrome", function(){
+      assert.strictEqual(isPalindrome("launchcode"), false);
+   });
+
+   it("should return false for a simple non-palindrome", function(){
+      assert.strictEqual(isPalindrome("ab"), false);
+   });
+
+   it("should be case-sensitive", function(){
+      assert.strictEqual(isPalindrome("abA"), false);
+   });
+
+   it("should consider whitespace", function(){
+      assert.strictEqual(isPalindrome("so many dynamos"), false);
+   });
+   
+Now run the tests to make sure they pass. Your code now includes a set of tests that considers a wide variety of positive and negative cases.
+
+Edge Cases
+----------
+
+Recall our definition of **edge case**:
+
+.. pull-quote::
+
+   An edge case is a test case that provides input at the extreme edge of what the unit should be able to handle.
+
+Edge cases can look very different for different units of code. Most of the examples we provided above dealt with numerical edge cases. However, edge cases can also be non-numeric. 
+
+In the case of ``isPalindrome``, the most obvious edge case would be that of the empty string, ``""``. This is the smallest possible string that we can use when calling ``isPalindrome``. Not only is it the smallest, but it is essentially *different* from the next longest string, ``"a"``---one has characters and one doesn't. 
+
+Should the empty string be considered a palindrome? That decision is up to us, the programmer, and there is not right or wrong answer. In our case, we have decided to take a very literal definition of the term "palindrome" by considering case and whitespace. In other words, our definition says that a string is a palindrome exactly when it equals its reverse. Since the reverse of ``""`` is also ``""``, it makes sense to consider the emtpy string a palindrome.
```

Comment:
Typo fix: "...there is no right or wrong answer."
Passive to active fix: "In our case, we decided to..."
Word removal: "Our definition says that a string is a palindrome when it equals its reverse."